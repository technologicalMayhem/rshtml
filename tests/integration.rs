use insta::assert_snapshot;
use rshtml::{convert_file, convert_directory};

#[test]
fn single_file() {
    let path_in = "./tests/input/single_file/index.html";

    let result = convert_file(path_in).unwrap();

    assert_snapshot!(result);
}

#[test]
fn multi_file() {
    let path_in = "./tests/input/multi_file";

    let result = convert_directory(path_in).unwrap();

    assert_snapshot!(result)
}