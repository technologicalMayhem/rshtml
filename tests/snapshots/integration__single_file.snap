---
source: tests/integration.rs
expression: result
---
pub struct Index {
    name: String,
    page_name: String,
}

impl Index {
    pub fn render(self) -> String {
        let name = self.name;
        let page_name = self.page_name;
        format!("<p>Hello, {name} welcome @{page_name}</p>")
    }

    pub fn new(name: String, page_name: String) -> Self {
        Self { name, page_name }
    }
}

